packages/contracts/contracts/interpreters/CoinTransferFromVirtualAppInterpreter.sol

```diff
@@ -0,0 +1,59 @@
+pragma solidity 0.5.10;
+pragma experimental "ABIEncoderV2";
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+
+import "../interfaces/Interpreter.sol";
+import "../libs/LibOutcome.sol";
+
+
+/// @notice 
+/// 
+/// Interprets a finalized AppInstance whose outcome is of type
+/// CoinTransfer. While CoinTransfer outcome contains `to` and `coinAddress`
+/// fields, those fields are *ignored* by this interpreter. Only the `amount`
+/// field is used.
+
+contract CoinTransferFromVirtualAppInterpreter is Interpreter {
+
+  struct Agreement {
+    uint256 capitalProvided;
+    uint256 expiryBlock;
+    address payable[2] beneficiaries;
+    address tokenAddress;
+  }
+
+  function interpretOutcomeAndExecuteEffect(
+    bytes calldata encodedOutcome,
+    bytes calldata params
+  )
+    external
+  {
+    LibOutcome.CoinTransfer memory outcome = abi.decode(
+      encodedOutcome,
+      (LibOutcome.CoinTransfer)
+    );
+
+    Agreement memory agreement = abi.decode(
+      params,
+      (Agreement)
+    );
+
+    require(
+      block.number < agreement.expiryBlock,
+      "Virtual App agreement has expired and is no longer valid."
+    );
+
+    ERC20(agreement.tokenAddress).transfer(
```

Comment:
`tokenAddress` might not be an ERC20 asset

Reply:
Ah, good catch.